modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2152,7 +2152,7 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
 
     # __getitem__ methods
     __getitem_bool = Binary.register(
-        lambda df, r: df[r], join_type="left", labels="drop"
+        lambda df, r: df[r] if r.size > 1 else df[[r]], join_type="left", labels="drop"
```

Comment:
Just for clarification, but basically what was happening is that we had `r` being a list of bools: `[False, True, False, ...]`? And if we had only one item then the array would get squeezed to a scalar?

Reply:
this would make a nice in-line comment, pls add

Reply:
this would make a nice in-line comment, pls add